### ISSUE TITLE
Feature: hide interfaces without IP address from several points from the GUI

### ISSUE THREAD

**Issue by 8191** [opnsense/core#584]
There are some use-cases where it makes sense to configure an enabled interface without an IP address, e.g. when the interface is part of a bridge and only the bridge has an IP. In that case, currently, OPNsense also shows interfaces without IP address to be chosen for many settings where it definitely makes no sense, e.g. interfaces the DNS Resolver should bind to, local subnet selector of IPSec P2 entries, etc... Also the Firewall: Rules view get's quite cluttered with all interfaces being part of a bridge, but typically do not require firewall rules (since the rules sit on the bridge, not the physical interface).

What about hiding interfaces without IP address generally from the GUI and adding a per-interface override option for that? Or maybe better, hide interfaces without IP from points where choosing the interfaces leads to an invalid configuration setting (e.g. IPSec P2 entry)?

**Comment by AdSchellevis** [comment]
Hi Manuel, 
I do like the idea of hiding interfaces which have no relevant use for the specific application, although I do want to minimize the number of major changes in the legacy code base leading to new (and possibly more complex) code.
For new mvc style code, filters can easily be defined within the xml model base, the old code has different kinds of magic for almost the same outcome (functions parsing config sections, or direct parsing within the ui page itself).

The interface section really is in need of improvement, the question is how to get there step by step without interfering too much with old code that is there already. For example the Aiccu support needs an interface (/ zone) which isn't managed by the system (no ifconfig kind of stuff), but can be managed from the components using the interface (/zone).
Coming to think about it, maybe we should add a scope attribute to the system which we define very thoroughly and make sure the backend code obeys that first so we can materialize all interface/zone types into a single structure and keep all rules around that clean. (ipsec/openvpn should in that case also manage their "interfaces" into the same structure).
Eventually we could/should move the interface structure into the new model system, but given the complexity of all behind it it's probably easier (and better) to cleanup functions in the old code first.

Long story short, if you want to filter some interfaces in separate pages to make it better readable, keeping the impact only on the affected user pages, I'm absolutely in favour of doing that. creating new functions which filters all use, rather not (now), unless we think of a strategy which can also work in the long run.

Cheers,

Ad

**Comment by AdSchellevis** [comment]
I guess its better to close this one, there have been quite some changes in the meantime.
